services/emailservice.js

```diff
@@ -0,0 +1,87 @@
+
+/* 
+Rudimentary email service
+Currently creates emails for account creation, password changes, and error message if something else is attempted.
+
+TODO:
+- replace gmail with SMTP server
+- incorporate userID, so a client's username/password/email can be fetched with a GET request
+------ or will this already be done in some other piece of code?
+- make transporter password a function parameter, instead of just having the email password sit there in plain sight
+------ transporter's fields would have to be created in an outside function
+------ then, just do:
+------ let transporter = nodemailer.createTransport(setTransporterFields(password))
+
+NOTES:
+- got rid of "await" that was in front of transporter.sendMail(mail);
+------ while the code successfully sends emails now, info.messageId now becomes undefined. 
+------ does this matter? is there some other reason why it's better to have await?
```

Comment:
transporter.sendMail returns a promise or promise-like object (not sure which, as I haven't used the library myself). This means that this method happens asynchronously, and the code written after this method call continues executing while the sendMail method continues executing in the background. Await is a special JS method that causes the code to "pause" until that value is returned (it really doesn't pause. it does some analysis to determine where that value is used and executes all code that it can until it reaches code that depends on the awaited function). We definitely want to use await, or some callback to ensure that we get information on whether the email was sent successfully.